### PR TITLE
Adding titles to CLI doc.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: docs/source/basic_tutorials/launcher.md [hunk 1]
```diff
@@ -135,57 +189,87 @@ Options:
...
+```
+#WEIGHTS_CACHE_OVERRIDE
+```shell
       --weights-cache-override <WEIGHTS_CACHE_OVERRIDE>
           The location of the huggingface hub cache. Used to override the location if you want to provide a mounted disk for instance
           
           [env: WEIGHTS_CACHE_OVERRIDE=]
 
+```
+#disable-custom-kernels
```
Comment (pcuenca) [inline docs/source/basic_tutorials/launcher.md]:
```suggestion
#DISABLE_CUSTOM_KERNELS
```

Just for consistency with the rest, not sure if the parsing code cares about it or not.

File: docs/source/basic_tutorials/launcher.md [hunk 2]
```diff
@@ -198,50 +282,86 @@ Options:
...
+```
+#ROPE_FACTOR
+```shell
       --rope-factor <ROPE_FACTOR>
           Rope scaling will only be used for RoPE models See `rope_scaling`
           
           [env: ROPE_FACTOR=]
 
+```
+#json-output
```
Comment (pcuenca) [inline docs/source/basic_tutorials/launcher.md]:
```suggestion
#JSON_OUTPUT
```

File: docs/source/basic_tutorials/launcher.md [hunk 3]
```diff
@@ -198,50 +282,86 @@ Options:
...
           [env: WATERMARK_GAMMA=]
 
+```
+#WATERMARK_DELTA
+```shell
       --watermark-delta <WATERMARK_DELTA>
           [env: WATERMARK_DELTA=]
 
+```
+#ngrok
```
Comment (pcuenca) [inline docs/source/basic_tutorials/launcher.md]:
```suggestion
#NGROK
```

File: docs/source/basic_tutorials/launcher.md [hunk 4]
```diff
@@ -198,50 +282,86 @@ Options:
...
+```
+#NGROK_EDGE
+```shell
       --ngrok-edge <NGROK_EDGE>
           ngrok edge
           
           [env: NGROK_EDGE=]
 
+```
+#env
```
Comment (pcuenca) [inline docs/source/basic_tutorials/launcher.md]:
```suggestion
#ENV
```

File: docs/source/basic_tutorials/launcher.md [hunk 5]
```diff
@@ -198,50 +282,86 @@ Options:
...
+```shell
   -e, --env
           Display a lot of information about your runtime environment
 
+```
+#help
```
Comment (pcuenca) [inline docs/source/basic_tutorials/launcher.md]:
```suggestion
#HELP
```

File: docs/source/basic_tutorials/launcher.md [hunk 6]
```diff
@@ -198,50 +282,86 @@ Options:
...
+```shell
   -h, --help
           Print help (see a summary with '-h')
 
+```
+#version
```
Comment (pcuenca) [inline docs/source/basic_tutorials/launcher.md]:
```suggestion
#VERSION
```

File: docs/source/basic_tutorials/launcher.md [hunk 7]
```diff
@@ -135,57 +189,87 @@ Options:
...
+```
+## WEIGHTS_CACHE_OVERRIDE
+```shell
       --weights-cache-override <WEIGHTS_CACHE_OVERRIDE>
           The location of the huggingface hub cache. Used to override the location if you want to provide a mounted disk for instance
           
           [env: WEIGHTS_CACHE_OVERRIDE=]
 
+```
+## disable-custom-kernels
```
Comment (pcuenca) [inline docs/source/basic_tutorials/launcher.md:254]:
```suggestion
## DISABLE_CUSTOM_KERNELS
```

Just for consistency, not sure it matters.

File: docs/source/basic_tutorials/launcher.md [hunk 8]
```diff
@@ -198,50 +282,86 @@ Options:
...
+```
+## ROPE_FACTOR
+```shell
       --rope-factor <ROPE_FACTOR>
           Rope scaling will only be used for RoPE models See `rope_scaling`
           
           [env: ROPE_FACTOR=]
 
+```
+## json-output
```
Comment (pcuenca) [inline docs/source/basic_tutorials/launcher.md:294]:
```suggestion
## JSON_OUTPUT
```

File: docs/source/basic_tutorials/launcher.md [hunk 9]
```diff
@@ -198,50 +282,86 @@ Options:
...
           [env: WATERMARK_GAMMA=]
 
+```
+## WATERMARK_DELTA
+```shell
       --watermark-delta <WATERMARK_DELTA>
           [env: WATERMARK_DELTA=]
 
+```
+## ngrok
```
Comment (pcuenca) [inline docs/source/basic_tutorials/launcher.md:326]:
```suggestion
## NGROK
```

File: docs/source/basic_tutorials/launcher.md [hunk 10]
```diff
@@ -198,50 +282,86 @@ Options:
...
+```
+## NGROK_EDGE
+```shell
       --ngrok-edge <NGROK_EDGE>
           ngrok edge
           
           [env: NGROK_EDGE=]
 
+```
+## env
```
Comment (pcuenca) [inline docs/source/basic_tutorials/launcher.md:350]:
```suggestion
## ENV
```

File: docs/source/basic_tutorials/launcher.md [hunk 11]
```diff
@@ -198,50 +282,86 @@ Options:
...
+```shell
   -e, --env
           Display a lot of information about your runtime environment
 
+```
+## help
```
Comment (pcuenca) [inline docs/source/basic_tutorials/launcher.md:356]:
```suggestion
## HELP
```

File: docs/source/basic_tutorials/launcher.md [hunk 12]
```diff
@@ -198,50 +282,86 @@ Options:
...
+```shell
   -h, --help
           Print help (see a summary with '-h')
 
+```
+## version
```
Comment (pcuenca) [inline docs/source/basic_tutorials/launcher.md:362]:
```suggestion
## VERSION
```

**Review by pcuenca — DISMISSED** [review]
Just suggested the same capitalization / underscoring for consistency, ignore if it's not relevant or appropriate :)

**Comment by Narsil** [comment]
@pcuenca this is all generated, it's easier to fix the script directly :)

File: update_doc.py [hunk 13]
```diff
@@ -32,17 +32,14 @@ def main():
                 header = tokens[-1][:-1]
             else:
                 header = line.split("--")[-1]
+            header = header.upper().replace("-", "_")
```
Comment (pcuenca) [inline update_doc.py:35]: 👍

**Pull request merged**